client/app/components/parameters.js

```diff
@@ -8,13 +8,45 @@ const ParameterSettingsComponent = {
     close: '&',
     dismiss: '&',
   },
-  controller() {
+  controller($sce, Query) {
     'ngInject';
 
+    this.trustAsHtml = html => $sce.trustAsHtml(html);
     this.parameter = this.resolve.parameter;
+    this.onQuerySelect = (query) => {
+      this.parameter.query = query;
```

Comment:
👍 

I think the putting only the id should be enough for the watch, if you encounter issues -- let me know and I will give a look.